File: frontend/src/services/policy.service.ts
import axios from "axios";

const api = axios.create({
  baseURL: "http://127.0.0.1:8000",
});

export type PolicyRequest = {
  currentAQI: number;
  traffic: number;
  industry: number;
  construction: number;
  greenery: number;
};

export type PolicyResponse = {
  currentAQI: number;
  predictedAQI: number;
  improvement: number;
  risk: string;
};

export async function predictPolicy(
  payload: PolicyRequest
): Promise<PolicyResponse> {
  const response = await api.post(
    "/policy/predict",
    payload
  );

  return response.data;
}

File: backend/app/services/policy_service.py
class PolicyService:

    @staticmethod
    def predict(
        current_aqi: int,
        traffic: int,
        industry: int,
        construction: int,
        greenery: int,
    ):

        reduction = (
            traffic * 0.35
            + industry * 0.40
            + construction * 0.30
            + greenery * 0.25
        )

        predicted = max(
            20,
            round(current_aqi - reduction),
        )

        improvement = round(
            ((current_aqi - predicted) / current_aqi) * 100,
            1,
        )

        if predicted <= 50:
            risk = "Good"
        elif predicted <= 100:
            risk = "Moderate"
        elif predicted <= 150:
            risk = "Unhealthy for Sensitive"
        elif predicted <= 200:
            risk = "Unhealthy"
        elif predicted <= 300:
            risk = "Very Unhealthy"
        else:
            risk = "Hazardous"

        return {
            "currentAQI": current_aqi,
            "predictedAQI": predicted,
            "improvement": improvement,
            "risk": risk,
        }